packages/store/addon/-private/system/record-notification-manager.ts

```diff
@@ -9,18 +9,27 @@ type UnsubscribeToken = Object;
 const Cache = new WeakMap<StableRecordIdentifier, NotificationCallback>();
 const Tokens = new WeakMap<UnsubscribeToken, StableRecordIdentifier>();
 
-interface NotificationCallback {
-  (
-    identifier: StableRecordIdentifier,
-    notificationType: 'attributes' | 'relationships' | 'identity' | 'errors' | 'meta' | 'unload' | 'property' | 'state'
-  ): void;
+export type NotificationType =
+  | 'attributes'
+  | 'relationships'
+  | 'identity'
+  | 'errors'
+  | 'meta'
+  | 'unload'
+  | 'property'
```

Comment:
Can we add a comment that this is only ED internal. We should also have a path away from it